### ISSUE TITLE
Normalization layer do not get unique names. Unable to save model.

### ISSUE THREAD

**Issue by antonstagge** [philipperemy/keras-tcn#121]
The normalization layers (BatchNorm or LayerNorm) do not get unique names. Since each ResidualBlock has 2 Norm layers, you will not be able to save the model.

It is an easy fix:
In tcn.py, line 100 add the line `with K.name_scope('norm_{}'.format(k)):`
currently looks like:
```
with K.name_scope(name):  # name scope used to make sure weights get unique names
    self._add_and_activate_layer(Conv1D(filters=self.nb_filters,
                                                        kernel_size=self.kernel_size,
                                                        dilation_rate=self.dilation_rate,
                                                        padding=self.padding,
                                                        name=name,
                                                        kernel_initializer=self.kernel_initializer))

if self.use_batch_norm:
    self._add_and_activate_layer(BatchNormalization())
elif self.use_layer_norm:
    self._add_and_activate_layer(LayerNormalization())
```
Change to:
```
with K.name_scope(name):  # name scope used to make sure weights get unique names
    self._add_and_activate_layer(Conv1D(filters=self.nb_filters,
                                                        kernel_size=self.kernel_size,
                                                        dilation_rate=self.dilation_rate,
                                                        padding=self.padding,
                                                        name=name,
                                                        kernel_initializer=self.kernel_initializer))
with K.name_scope('norm_{}'.format(k)):
    if self.use_batch_norm:
        self._add_and_activate_layer(BatchNormalization())
    elif self.use_layer_norm:
        self._add_and_activate_layer(LayerNormalization())
```

**Comment by philipperemy** [comment]
Thanks for reporting. Going to update that!

**Comment by philipperemy** [comment]
Merged.
